src/backend/binaryen/utils.ts

```diff
@@ -1816,23 +1869,128 @@ export namespace FunctionalFuncs {
                 }
                 break;
             }
-            default:
-                if (rightValueType.kind === ValueTypeKind.NUMBER) {
-                    res = operateF64F64ToDyn(
-                        module,
-                        leftValueRef,
-                        rightValueRef,
-                        opKind,
-                        true,
+            case ts.SyntaxKind.LessThanEqualsToken:
+            case ts.SyntaxKind.LessThanToken:
+            case ts.SyntaxKind.GreaterThanEqualsToken:
+            case ts.SyntaxKind.GreaterThanToken: {
```

Comment:
Seems that these four case situation is fully included in the `default` statement? How about remove these four case statement, and ensure the default statement is handled correctly?